### DISCUSSION TITLE
zpool ashift value use more than 16

### DISCUSSION THREAD

**Discussion by httaotao** [openzfs/zfs#14543] [Ideas]
Hi, I am now using zfs dataset to store some data. However I find the zpool ashift max value is 16.

However, in my case, most of the files size are more than 100GB,  it's so large and i think ashift is too small to use. 

So in the future, can we use more than 16 of ashift in zpool ? 

thanks.

**Comment by robn** [comment]
`ashift` should match the block size of the underlying storage devices, which is why its almost always `9` (512B) or `12` (4KB). You almost certainly don't want any other values unless you're on some very unconventional hardware or otherwise doing something unusual.

You may however be thinking of the `recordsize` parameter. The default for this is 128KB, but if you're storing mostly very large files, you can set it as high as 1MB. This isn't a limit though; it just helps ZFS decide when to split large files into smaller pieces for storage. Tuning it for your workload can help with performance and drive wear but most of the time if you're just storing files (even large ones!) and everything seems fine, the default 128KB should be good enough.

This is a good article on both `ashift` and `recordsize`: [Tuning Recordsize in OpenZFS](https://klarasystems.com/articles/tuning-recordsize-in-openzfs/)

**Reply by httaotao** [reply]
ok, thanks for help
